### PR TITLE
kie-issues#467: Upgrade prettier from `2.x.x` to `3.x.x` on kie-tools 

### LINKED ISSUE #467: Upgrade `prettier` from `2.x.x` to `3.x.x` on `kie-tools`
This will allow KIE Tools to have multiple `.gitignore` files. See https://prettier.io/blog/2023/07/05/3.0.0.html#ignore-gitignored-files-by-default-14731httpsgithubcomprettierprettierpull14731-by-fiskerhttpsgithubcomfisker

### PR DESCRIPTION
**Pull request by ljmotta**
Closes: https://github.com/apache/incubator-kie-issues/issues/467

### pnpm-lock.yaml
This PR removes the `pnpm-lock.yaml` auto formating.

### VS Code extension
Troubleshooting the Prettier VS Code extension:
 - Update the Prettier VS Code extension;
 - Change the config file to `prettier.config.mjs` on the settings;
 - Disable/enable the VS Code extension;

### prettier 2.x.x
Some packages uses the `prettier.format` function. Starting from version `3.x.x` the `prettier` libraries uses ESM instead of CJS, making it incompatible with `jest` `29.x`. Due to this incompatibility issue, an execption for the `prettier` package was added to the `.syncpackrc.json` file, enabling two different versions of `prettier` on the monorepo.

### Change summary
- Add trailling comma to all object and types;
- Add line break on `package.json` end of file;
- Reorder commands on `package.json`. Colon has priority over dash;
- Format `svg` files;
- Format `scss` files. Change one-liner properties to multi-line;
- Format `css` files. Change one-liner properties to multi-line;
- Format `xml`, `xsd`, `dmn`, `bpmn`, `bpmn2`, `scesim` and any other `xml` compatible file
- Changed `DOCTYPE` to `doctype`;
- Ternary identantion;
- Class constructor changed to multi-line instead of one liner;

### CODE REVIEW

**Comment by tiagobento** [comment]
Hi @ljmotta, thank you so much for addressing this.

In addition to reviewing the `,` at the end of function calls that I mentioned to you in private chat, can you please also split this PR's commits in
1. The code base changes
2. Formatting with the new version of Prettier?

This will make reviewing much easier, as right now it's kind of impossible to go through all the changes that this PR brings.

File: packages/sonataflow-operator/package.json [hunk 1]
```diff
@@ -22,17 +22,17 @@
     "bump-version": "run-script-os",
     "bump-version:darwin:linux": "./hack/bump-version.sh",
     "bump-version:win32": "echo 'Bumping version not supported on Windows'",
-    "format": "prettier --write .",
+    "format": "prettier --write . --ignore-path=../../.prettierignore --ignore-path=../../.gitignore",
```
Comment (tiagobento) [inline packages/sonataflow-operator/package.json:25]: There are other packages doing the same thing, but you didn't add the `--ignore-path` argument to them. Why?

File: .prettierignore
Comment (tiagobento) [inline .prettierignore:1]: Why ignore XSDs?

File: .syncpackrc.json [hunk 2]
```diff
@@ -1 +1,12 @@
...
+    {
+      "dependencies": ["prettier"],
+      "packages": [
+        "@kie-tools/uniforms-bootstrap4-codegen",
+        "@kie-tools/uniforms-patternfly-codegen",
+        "@kie-tools/yaml-language-server"
+      ]
+    }
+  ]
+}
```
Comment (tiagobento) [inline .syncpackrc.json:12]: Would be great to make those already be mapped by an issue, so we don't loose track of this version discrepancy.

File: package.json [hunk 3]
```diff
@@ -23,23 +23,18 @@
     "@kie-tools-scripts/update-kogito-version": "workspace:*",
     "@kie-tools-scripts/update-stream-name": "workspace:*",
     "@kie-tools-scripts/update-version": "workspace:*",
-    "@nice-move/prettier-plugin-package-json": "^0.6.1",
-    "@prettier/plugin-xml": "^2",
+    "@nice-move/prettier-plugin-package-json": "^0.8.0",
+    "@prettier/plugin-xml": "^3.4.1",
```
Comment (tiagobento) [inline package.json:27]: Still no change do XMLs... I wonder if this is working, to be honest. Maybe we leave that for

- https://github.com/apache/incubator-kie-issues/issues/1156
 
?

**Review by tiagobento — CHANGES_REQUESTED** [review]

File: packages/sonataflow-operator/package.json (hunk 1 above)
Comment (ljmotta) [reply packages/sonataflow-operator/package.json:25]: I still need to revisit all packages, running the format on the package folder don't take into account the `.prettierignore` on the monorepo root, which causes the formatting to break

File: .prettierignore
Comment (ljmotta) [reply .prettierignore:1]: Prettier is wrongfully formatting XSDs, but I'll add a rule to format it as a `xml` file.

File: .syncpackrc.json (hunk 2 above)
Comment (ljmotta) [reply .syncpackrc.json:12]: Sure!

File: .prettierignore
Comment (ljmotta) [reply .prettierignore:1]: actually, it wasn't... @prettier/parser-xml doesn't support DTD!

File: package.json (hunk 3 above)
Comment (ljmotta) [reply package.json:27]: It's working, just not identing, I'll change that.

File: .syncpackrc.json (hunk 2 above)
Comment (tiagobento) [reply .syncpackrc.json:12]: https://github.com/apache/incubator-kie-issues/issues/1332

**Comment by tiagobento** [comment]
Thanks! Love to see the XMLs properly indented 🙏

**Review by tiagobento — APPROVED** [review]

**Review by thiagoelg — APPROVED** [review]

**Review by tiagobento — APPROVED** [review]

**Pull request merged**